express/blocks/shared/floating-cta.js

```diff
@@ -0,0 +1,356 @@
+  });
+
+  const linksPopulated = new CustomEvent('linkspopulated', { detail: [$floatButtonLink, $lottieScrollButton] });
+  document.dispatchEvent(linksPopulated);
+
+  $floatButtonInnerWrapper.append($floatButtonBackground, $floatButtonLink);
+  $floatButton.append($floatButtonInnerWrapper, $lottieScrollButton);
+  $floatButtonWrapper.append($floatButton);
+  main.append($floatButtonWrapper);
+  if ($floatButtonWrapperOld) {
+    const $parent = $floatButtonWrapperOld.parentElement;
+    if ($parent && $parent.children.length === 1) {
+      $parent.remove();
+    } else {
+      $floatButtonWrapperOld.remove();
+    }
+  }
+
+  // Floating button scroll/click events
+  lazyLoadLottiePlayer();
+  const $scrollAnchor = document.querySelector('.section:not(:nth-child(1)):not(:nth-child(2)) .template-list, .section:not(:nth-child(1)):not(:nth-child(2)) .layouts, .section:not(:nth-child(1)):not(:nth-child(2)) .steps-highlight-container') ?? document.querySelector('.section:nth-child(3)');
+  if (!$scrollAnchor) {
+    hideScrollArrow($floatButtonWrapper, $lottieScrollButton);
+  } else {
+    initLottieArrow($lottieScrollButton, $floatButtonWrapper, $scrollAnchor, data);
+  }
+
+  // Intersection observer - hide button when scrolled to footer
+  const $footer = document.querySelector('footer');
+  if ($footer) {
+    const hideButtonWhenFooter = new IntersectionObserver((entries) => {
+      const entry = entries[0];
+      if (entry.intersectionRatio > 0 || entry.isIntersecting) {
+        $floatButtonWrapper.classList.add('floating-button--hidden');
+      } else {
+        $floatButtonWrapper.classList.remove('floating-button--hidden');
+      }
+    }, {
+      root: null,
+      rootMargin: '32px',
+      threshold: 0,
+    });
+
+    if (document.readyState === 'complete') {
+      hideButtonWhenFooter.observe($footer);
+    } else {
+      window.addEventListener('load', () => {
+        hideButtonWhenFooter.observe($footer);
+      });
+    }
+  }
+
+  const $heroCTA = document.querySelector('a.button.same-as-floating-button-CTA');
+  if ($heroCTA) {
+    const hideButtonWhenIntersecting = new IntersectionObserver((entries) => {
+      const $e = entries[0];
+      if ($e.boundingClientRect.top > window.innerHeight - 40 || $e.boundingClientRect.top === 0) {
+        $floatButtonWrapper.classList.remove('floating-button--below-the-fold');
+        $floatButtonWrapper.classList.add('floating-button--above-the-fold');
+      } else {
+        $floatButtonWrapper.classList.add('floating-button--below-the-fold');
+        $floatButtonWrapper.classList.remove('floating-button--above-the-fold');
+      }
+      if ($e.intersectionRatio > 0 || $e.isIntersecting) {
+        $floatButtonWrapper.classList.add('floating-button--intersecting');
+      } else {
+        $floatButtonWrapper.classList.remove('floating-button--intersecting');
+      }
+    }, {
+      root: null,
+      rootMargin: '-40px 0px -40px 0px',
```

Comment:
```suggestion
      rootMargin: '-40px 0',
```

Reply:
I shortened it to 2 attributes but the 'px' turns out to be required for this case. Otherwise I see an error in the console.

Reply:
Weird...